src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIMetadataService.java

```diff
@@ -68,10 +90,15 @@ public MetadataTemplate createMetadataTemplate(MetadataTemplate template, Long p
 	 * @throws Exception if there is an error updating the template
 	 */
 	public String updateMetadataTemplate(MetadataTemplate template, Locale locale) throws Exception {
```

Comment:
This method needs to be wrapped with `@Transactional`, otherwise it will have a detrimental performance when updating the metadata template on a project with a large number of samples.

Reply:
Added in c427089ae2f1a86e46a20ce76e32f226b2ca3b89